src/providers/hana/qgshanaresultset.cpp

```diff
@@ -0,0 +1,142 @@
+}
+
+QVariant QgsHanaResultSet::getValue( unsigned short columnIndex )
+{
+  switch ( mMetadata->getColumnType( columnIndex ) )
+  {
+    case SQLDataTypes::Bit:
+    case SQLDataTypes::Boolean:
+      return QgsHanaUtils::toVariant( mResultSet->getBoolean( columnIndex ), QVariant::Bool );
+    case SQLDataTypes::Char:
+      return QgsHanaUtils::toVariant( mResultSet->getByte( columnIndex ) );
+    case SQLDataTypes::WChar:
+      return QgsHanaUtils::toVariant( mResultSet->getUByte( columnIndex ) );
+    case SQLDataTypes::SmallInt:
+      if ( mMetadata ->isSigned( columnIndex ) )
+        return QgsHanaUtils::toVariant( mResultSet->getShort( columnIndex ) );
+      else
+        return QgsHanaUtils::toVariant( mResultSet->getUShort( columnIndex ) );
+    case SQLDataTypes::Integer:
+      if ( mMetadata ->isSigned( columnIndex ) )
+        return QgsHanaUtils::toVariant( mResultSet->getInt( columnIndex ) );
+      else
+        return QgsHanaUtils::toVariant( mResultSet->getUInt( columnIndex ) );
+    case SQLDataTypes::BigInt:
+      if ( mMetadata ->isSigned( columnIndex ) )
+        return QgsHanaUtils::toVariant( mResultSet->getLong( columnIndex ) );
+      else
+        return QgsHanaUtils::toVariant( mResultSet->getULong( columnIndex ) );
+    case SQLDataTypes::Double:
+    case SQLDataTypes::Decimal:
+    case SQLDataTypes::Numeric:
+      return QgsHanaUtils::toVariant( mResultSet->getDouble( columnIndex ) );
+    case SQLDataTypes::Date:
+    case SQLDataTypes::TypeDate:
+      return QgsHanaUtils::toVariant( mResultSet->getDate( columnIndex ) );
+    case SQLDataTypes::Time:
+    case SQLDataTypes::TypeTime:
+      return QgsHanaUtils::toVariant( mResultSet->getTime( columnIndex ) );
+    case SQLDataTypes::Timestamp:
+    case SQLDataTypes::TypeTimestamp:
+      return QgsHanaUtils::toVariant( mResultSet->getTimestamp( columnIndex ) );
+    case SQLDataTypes::VarChar:
+    case SQLDataTypes::LongVarChar:
+      return QgsHanaUtils::toVariant( mResultSet->getString( columnIndex ) );
+    case SQLDataTypes::WVarChar:
+    case SQLDataTypes::WLongVarChar:
+      return QgsHanaUtils::toVariant( mResultSet->getNString( columnIndex ) );
+    case SQLDataTypes::Binary:
+    case SQLDataTypes::VarBinary:
+    case SQLDataTypes::LongVarBinary:
+      return QgsHanaUtils::toVariant( mResultSet->getBinary( columnIndex ) );
+    default:
+      QgsDebugMsg( QStringLiteral( "Unhandled HANA type %1" ).arg( QString::fromStdString( mMetadata->getColumnTypeName( columnIndex ) ) ) );
+      return QVariant();
+  }
+}
+
+QgsGeometry QgsHanaResultSet::getGeometry( unsigned short columnIndex )
+{
+  size_t bufLength = mResultSet->getBinaryLength( columnIndex );
+  unsigned char *bufPtr = nullptr;
+
+  if ( bufLength == ResultSet::UNKNOWN_LENGTH )
+  {
+    Binary wkb = mResultSet->getBinary( columnIndex );
+    if ( wkb.isNull() || wkb->size() == 0 )
+      bufLength = 0;
+    else
+      bufPtr = reinterpret_cast<unsigned char *>( wkb->data() );
```

Comment:
You also need to set `bufLength` here. Otherwise we will use `ResultSet::UNKNOWN_LENGTH` below.

Reply:
Done in 4d36f6f.